test/foundry/BaseSetup.t.sol

```diff
@@ -17,40 +17,129 @@ import {Currency} from '@aave/lens-protocol/contracts/mocks/Currency.sol';
 import {NFT} from 'contracts/mocks/NFT.sol';
 
 contract BaseSetup is Test {
-    uint256 constant firstProfileId = 1;
-    address constant deployer = address(1);
-    address constant governance = address(2);
-    address constant treasury = address(3);
+    using stdJson for string;
+
+    string forkEnv;
+    bool fork;
+    string network;
+    string json;
+    uint256 forkBlockNumber;
+
+    // TODO: Consider refactoring back to immutable + ternary
+    uint256 firstProfileId;
+    address deployer;
+    address governance;
+    address treasury;
+
     address constant publisher = address(4);
     address constant user = address(5);
     address constant userTwo = address(6);
     address constant userThree = address(7);
     address constant userFour = address(8);
     address constant userFive = address(9);
-    address immutable me = address(this); // Main test User is this (rather inheriting from this) contract
+    address immutable me = address(this);
 
-    string constant MOCK_HANDLE = 'handle.lens';
+    string constant MOCK_HANDLE = 'mock';
     string constant MOCK_URI = 'ipfs://QmUXfQWe43RKx31VzA2BnbwhSMW8WuaJvszFWChD59m76U';
     string constant OTHER_MOCK_URI =
         'https://ipfs.io/ipfs/QmTFLSXdEQ6qsSzaXaCSNtiv6wA56qq87ytXJ182dXDQJS';
     string constant MOCK_FOLLOW_NFT_URI =
         'https://ipfs.io/ipfs/QmU8Lv1fk31xYdghzFrLm6CiFcwVg7hdgV6BBWesu6EqLj';
 
-    uint16 constant TREASURY_FEE_BPS = 50;
+    uint16 TREASURY_FEE_BPS;
     uint16 constant TREASURY_FEE_MAX_BPS = 10000;
 
-    address immutable hubProxyAddr;
-    CollectNFT immutable collectNFT;
-    FollowNFT immutable followNFT;
-    LensHub immutable hubImpl;
-    TransparentUpgradeableProxy immutable hubAsProxy;
-    LensHub immutable hub;
-    FreeCollectModule immutable freeCollectModule;
-    Currency immutable currency;
-    ModuleGlobals immutable moduleGlobals;
-    NFT immutable nft;
+    address hubProxyAddr;
+    CollectNFT collectNFT;
+    FollowNFT followNFT;
+    LensHub hubImpl;
+    TransparentUpgradeableProxy hubAsProxy;
+    LensHub hub;
+    FreeCollectModule freeCollectModule;
+    Currency currency;
+    ModuleGlobals moduleGlobals;
+    NFT nft;
+
+    // TODO: Replace with forge-std/StdJson.sol::keyExists(...) when/if this PR is approved:
+    //       https://github.com/foundry-rs/forge-std/pull/226
+    function keyExists(string memory key) internal view returns (bool) {
+        return json.parseRaw(key).length > 0;
+    }
+
+    // TODO: Consider refactoring these funcs into a library (similar functions are also used in deploy script):
```

Comment:
Not a hard requirement but I think this is a good idea, less potential room for mistakes down the line!